step-generation/src/utils/misc.ts

```diff
@@ -1382,3 +1382,12 @@ export const labwareMatchesLabwareInHopper = (
   const labwareToBeStored = def.parameters.loadName
   return loadedLabware === labwareToBeStored
 }
+
+export const spaceInHopper = (
+  stackerState: FlexStackerModuleState | null
+): boolean => {
+  const maximumAllowedLabware = stackerState?.maxPoolCount ?? 0
+  const labwareStored = stackerState?.labwareInHopper
+  const numberOfLabwareStored = labwareStored?.length ?? 0
+  return numberOfLabwareStored + 1 < maximumAllowedLabware
```

Comment:
i think we don't need this `+1`. We're checking if there is space in the hopper at all at this point. With the +1, you could add 1 too many labwares, right?
```suggestion
  return numberOfLabwareStored < maximumAllowedLabware
```

Reply:
oh yes you are right over thought this one!